usbsdmux/__main__.py

```diff
@@ -68,6 +77,19 @@ def main():
         print(f"Does {args.sg} really point to an USB-SD-Mux?")
         sys.exit(1)
     mode = args.mode.lower()
+    gpio_action = args.gpio_action.lower()
+
+    if mode in ("gpio0", "gpio1"):
+        # Preflight checks for GPIO mode
+        if not ctl.type == UsbSdMuxFactory.SD_MUX_TYPE_FAST:
+            print("This USB-SD-Mux does not support GPIOs.", file=sys.stderr)
+            exit(1)
+        if not gpio_action:
+            gpio_action = "get"
+        if mode == "gpio0":
+            gpio = UsbSdMuxFast.gpio0
+        elif mode == "gpio1":
+            gpio = UsbSdMuxFast.gpio1
```

Comment:
`main()` shouldn't need to know about the details of the type. Just try to call methods defined on the base class and handle `NotImplemented` exceptions.